### PR TITLE
Refactor the main menu code (part 3)

### LINKED ISSUE #19311: Left/Right sound triggers only function on main menu
The main menu's [left/right movements](https://github.com/CleverRaven/Cataclysm-DDA/blob/ede2ddb8887daa551c911f62cc46b5495295747b/src/main_menu.cpp#L387-L394) sound triggers currently only occur on base menu items.

That is to say, secondary menu items do not have a noise triggered. E.g. opening "Special" from the home screen and then moving between "Defense" and "Tutorial."

### PR DESCRIPTION
**Pull request by sethsimon**
This is a continuation of #19094. This also fixes #19311.

### CODE REVIEW

File: src/main_menu.cpp [hunk 1]
```diff
@@ -1006,3 +887,141 @@ bool main_menu::new_character_tab()
...
+            //hide savegames if lua is not available for a lua-built world
+            if( ( wn != "TUTORIAL" && wn != "DEFENSE" ) && world_generator->world_need_lua_build( wn ) ) {
+                savegames.clear();
+                mvwprintz( w_open, iMenuOffsetY - 2, 15 + iMenuOffsetX + extra_w / 2,
+                           c_red, _( "This world requires the game to be compiled with Lua." ) );
+            } else if( savegames.empty() ) {
+                mvwprintz( w_open, iMenuOffsetY - 2, 19 + 19 + iMenuOffsetX + extra_w / 2,
+                           c_red, _( "No save games found!" ) );
+            } else {
+                for( std::vector<std::string>::iterator it = savegames.begin();
```
Comment (illi-kun) [inline src/main_menu.cpp]: I guess that ``` for( auto it : savegames )``` do the same but seems to be simpler to read.

File: src/main_menu.h [hunk 2]
```diff
@@ -24,8 +25,14 @@ class main_menu
         std::vector<std::string> vMenuItems; // MOTD, New Game, Load Game, etc.
         std::vector< std::vector<std::string> > vMenuHotkeys; // hotkeys for the vMenuItems
 
+        // Play a sound whenver the user moves left or right in the main menu or its tabs
```
Comment (illi-kun) [inline src/main_menu.h]: Why do you need this in header?

**Comment by Coolthulhu** [comment]
Works, but needs some cleaning up - that `sounds.h` is totally unnecessary in `main_menu.h`. Functions should only be implemented in headers if it's important for execution speed.

File: src/main_menu.h (hunk 2 above)
Comment (sethsimon) [inline src/main_menu.h]: It doesn't need to be implemented there. I just have a habit of implementing short functions in the header. I'd be glad to move it.

File: src/main_menu.cpp (hunk 1 above)
Comment (sethsimon) [inline src/main_menu.cpp]: Except there a couple places within the loop that use `it` as opposed to `*it`. But most of that code can definitely be simplified by using STL functions.

**Comment by sethsimon** [comment]
This is ready.

**Comment by illi-kun** [comment]
Astyle regression.

**Comment by sethsimon** [comment]
Fixed.

**Pull request merged**